app/components/anchor-wallet-provider.tsx

```diff
@@ -33,3 +37,12 @@ export function AnchorWalletProvider({ children }: AnchorWalletProviderProps) {
     </ConnectionProvider>
   );
 }
+
+export function useAnchorProvider() {
+  const { connection } = useConnection();
+  const wallet = useWallet();
+
+  return new AnchorProvider(connection, wallet as AnchorWallet, {
```

Comment:
I thought we had to do this, but another post I found said it was not needed!?

Sorry about that, glad you found it!

Reply:
Yes, some people said using anchor in the UI is not recommended, but I saw several examples that use it.